services/checks/checks.go

```diff
@@ -927,6 +1046,18 @@ func (s *Service) collectChecks(ctx context.Context) {
 	}
 
 	checks = s.filterSupportedChecks(checks)
+	for i, c := range checks {
+		cs, err := models.FindCheckStateByName(s.db.Querier, c.Name)
```

Comment:
Wouldn't it better to just get all `CheckStates` and convert to map? to not have n queries to DB.

Reply:
done